### PR TITLE
[#5594] Platform:Handle invalid certs/keys correctly.

### PR DESCRIPTION
**Pull request by jitendra-12113**
[5594][Platform] Handle invalid certs/keys correctly
**Description:**
We wanted to verify the cert at upload time itself.
**Testing:**
When we try to upload selfSigned cert, we were able to upload any
garbage files. Now with this fix, we have handled this scenrio. User
won't be able to upload garbage file. Incase of self signed cert upload
workflow, We are matching the modulus of
cert RSA Public key with the modulus of its private key if its matched
then it means certs are valid otheriwse Invalid.
For CA Cert, we are just validating the Cert format.
Added the unit Tests as well.

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java [hunk 1]
```diff
@@ -390,4 +376,87 @@ public static void createChecksums() {
...
+    File keyFile = new File(keyPath);
+    try (JcaPEMWriter keyWriter = new JcaPEMWriter(new FileWriter(keyFile))) {
+      keyWriter.writeObject(keyContent);
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      throw new RuntimeException("Save privateKey failed.");
+    }
+  }
+
+  public static void saveCertContent(X509Certificate cert, String certPath) {
```
Comment (Arnav15) [inline managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java:416]: Rename similar to `writeKeyFileContentToKeyPath`

**Review by Arnav15 — CHANGES_REQUESTED** [review]
Looks good overall. Just few minor comments.
1) Add a unit test for a bad cert case as well.
2) Revert the submodule changes. You can do so by running `git submodule update --init --recursive`

**Comment by Arnav15** [comment]
Also, can you update the description a little more. Exactly mention on how we are solving the issue and what all changes are made as part of this PR. And add the output of `sbt test`

**Comment by jitendra-12113** [comment]
Created separated PR for this:
https://github.com/yugabyte/yugabyte-db/pull/7264

**Pull request closed without merging**